ci/infra/testrunner/tests/test_istio_service_mesh.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import pytest
+import time
+
+CLUSTERROLEBINDING = ("""
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: istio-test
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: suse:caasp:psp:privileged
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: default
+- kind: ServiceAccount
+  name: httpbin
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-productpage
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-reviews
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-ratings 
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-details
+  namespace: default
+- kind: ServiceAccount
+  name: sleep
+  namespace: default
+EOF
+""")
+
+ISTIO_VERSION = "1.5"
+ISTIO_VERSION_PATCH = ISTIO_VERSION + ".4"
+ISTIO_URL = "https://raw.githubusercontent.com/istio/istio/release-" + ISTIO_VERSION + "/samples"
+
+def _istio_bookinfo_setup(kubectl):
+    istioctl = ("""
+                istioctl --kubeconfig={config} manifest apply \
+                         --set profile=default \
+                         --set addonComponents.prometheus.enabled=false \
+                         --set hub=registry.suse.de/devel/caasp/4.5/containers/containers/caasp/v4.5 \
+                         --set tag={version} \
+                         --set values.pilot.image=istio-pilot \
+                         --set values.global.proxy.image=istio-proxyv2 \
+                         --set values.global.proxy_init.image=istio-proxyv2
+                 """.format(config=kubectl.get_kubeconfig(), version=ISTIO_VERSION_PATCH))
+
+    kubectl.utils.runshellcommand(istioctl)
+    kubectl.run_kubectl("-n istio-system wait --for=condition=available deploy/istio-ingressgateway --timeout=3m")
+
+    # Activate service in namespace default
+    kubectl.run_kubectl("label namespace default istio-injection=enabled --overwrite")
+
+    # Create the clusterrolebinding to run everything
+    kubectl.run_kubectl("apply -f - << EOF " + CLUSTERROLEBINDING)
+
+    # Deploy bookinfo application
+    kubectl.run_kubectl("apply -f {manifest_url}/bookinfo/platform/kube/bookinfo.yaml".format(manifest_url=ISTIO_URL))
```

Comment:
I you want to further simplify the code you could use the f-string for formatting:
```
    kubectl.run_kubectl(f'apply -f {ISTIO_URL}/bookinfo/platform/kube/bookinfo.yaml')
```

Reply:
I did not know about this way of substituting variables in a string. Let me do that, looks better